src/Microsoft.AspNet.Mvc.Core/Formatters/AcceptHeaderParser.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Globalization;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNet.Mvc.Formatters
+{
+    public static class AcceptHeaderParser
+    {
+        private static readonly StringSegment QualityParameter = new StringSegment("q");
+
+        public static IList<MediaTypeSegmentWithQuality> ParseAcceptHeader(IList<string> acceptHeaders)
+        {
+            var parsedValues = new List<MediaTypeSegmentWithQuality>();
+            ParseAcceptHeader(acceptHeaders, parsedValues);
+
+            return parsedValues;
+        }
+
+        public static void ParseAcceptHeader(IList<string> acceptHeaders, IList<MediaTypeSegmentWithQuality> parsedValues)
+        {
+            if (acceptHeaders == null)
+            {
+                throw new ArgumentNullException(nameof(acceptHeaders));
+            }
+
+            if (parsedValues == null)
+            {
+                throw new ArgumentNullException(nameof(parsedValues));
+            }
+
+            for (var i = 0; i < acceptHeaders.Count; i++)
+            {
+                var index = 0;
+                var value = acceptHeaders[i];
+
+                while (!string.IsNullOrEmpty(value) && index < value.Length)
+                {
+                    MediaTypeSegmentWithQuality output;
+                    if (TryParseValue(value, ref index, out output))
+                    {
+                        // The entry may not contain an actual value, like " , "
+                        if (output.MediaType.HasValue)
+                        {
+                            parsedValues.Add(output);
+                        }
+                    }
+                    else
+                    {
+                        throw new FormatException(string.Format(CultureInfo.InvariantCulture, "Invalid values '{0}'.",
+                            value.Substring(index)));
+                    }
+                }
+            }
+        }
+
+        private static bool TryParseValue(string value, ref int index, out MediaTypeSegmentWithQuality parsedValue)
+        {
+            parsedValue = default(MediaTypeSegmentWithQuality);
+
+            // If multiple values are supported (i.e. list of values), then accept an empty string: The header may
+            // be added multiple times to the request/response message. E.g.
+            //  Accept: text/xml; q=1
+            //  Accept:
+            //  Accept: text/plain; q=0.2
```

Comment:
Multiple values refers to the Accept header appearing multiple times on the request response,
Accept: text/xml
Accept: application/json

is equivalent to
Accept: text/xml, application/json

I've changed the comment to clarify that it means multiple Accept headers appearing on the request